gpsd: does not compile with glibc because "-lm" missing?

Maintainer: @psidhu 
Environment: x86_64 glibc

Description:

Looks like the linking of the math library is not working?

```
x86_64-openwrt-linux-gnu-gcc -o gpsd -Wl,-z,now -Wl,-z,relro --sysroot=/tmp/openwrt/staging_dir/target-x86_64_glibc dbusexport.o gpsd.o shmexport.o timehint.o sd_socket.o -L. -L/tmp/openwrt/staging_dir/target-x86_64_glibc/usr/lib -L/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/usr/lib -L/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib -lgpsd -lgps_static -lusb-1.0
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: warning: libpthread.so.0, needed by /tmp/openwrt/staging_dir/target-x86_64_glibc/usr/lib/libusb-1.0.so, not found (try using -rpath or -rpath-link)
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: ./libgpsd.a(libgpsd_core.o): in function `fill_dop':
libgpsd_core.c:(.text+0xdbb): undefined reference to `sqrt'
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: libgpsd_core.c:(.text+0xdca): undefined reference to `sqrt'
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: libgpsd_core.c:(.text+0xdde): undefined reference to `sqrt'
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: libgpsd_core.c:(.text+0xded): undefined reference to `sqrt'
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: libgpsd_core.c:(.text+0xe01): undefined reference to `sqrt'
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: ./libgpsd.a(libgpsd_core.o):libgpsd_core.c:(.text+0xe0c): more undefined references to `sqrt' follow
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: ./libgpsd.a(libgpsd_core.o): in function `gpsd_poll'
```

Patching the scons file will result in

```
x86_64-openwrt-linux-gnu-gcc -o timespec_str.o -c --sysroot=/tmp/openwrt/staging_dir/target-x86_64_glibc -pthread -Wall -Wcast-align -Wextra -Wimplicit-fallthrough -Wmissing-declarations -Wmissing-prototypes -Wno-missing-field-initializers -Wno-uninitialized -Wpointer-arith -Wreturn-type -Wstrict-prototypes -Wvla -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -fmacro-prefix-map=/tmp/openwrt/build_dir/target-x86_64_glibc/gpsd-3.21=gpsd-3.21 -Wformat -Werror=format-security -fstack-protector -znow -zrelro -O2 -pthread -D_FORTIFY_SOURCE=1 -I/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/usr/include -I/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/include timespec_str.c
os.chdir('.')
x86_64-openwrt-linux-gnu-ar rc libgps_static.a ais_json.o bits.o gpsdclient.o gps_maskdump.o gpsutils.o hex.o json.o libgps_core.o libgps_dbus.o libgps_json.o libgps_shm.o libgps_sock.o netlib.o os_compat.o rtcm2_json.o rtcm3_json.o shared_json.o timespec_str.o
os.chdir('/tmp/openwrt/build_dir/target-x86_64_glibc/gpsd-3.21')
os.chdir('.')
x86_64-openwrt-linux-gnu-ranlib libgps_static.a
os.chdir('/tmp/openwrt/build_dir/target-x86_64_glibc/gpsd-3.21')
x86_64-openwrt-linux-gnu-gcc -o gpsd -Wl,-z,now -Wl,-z,relro --sysroot=/tmp/openwrt/staging_dir/target-x86_64_glibc -pthread dbusexport.o gpsd.o shmexport.o timehint.o sd_socket.o -L. -L/tmp/openwrt/staging_dir/target-x86_64_glibc/usr/lib -L/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/usr/lib -L/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib -lgpsd -lgps_static -lusb-1.0 -lm -lrt
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: cannot find /lib/libm.so.6
/tmp/openwrt/staging_dir/toolchain-x86_64_gcc-8.4.0_glibc/lib/gcc/x86_64-openwrt-linux-gnu/8.4.0/../../../../x86_64-openwrt-linux-gnu/bin/ld: cannot find /lib/libmvec.so.1
collect2: error: ld returned 1 exit status
scons: *** [gpsd] Error 1
scons: building terminated because of errors.
WARNING: asciidoctor not found.
WARNING: Some documentation and html will not be built.
make[2]: *** [Makefile:170: /tmp/openwrt/build_dir/target-x86_64_glibc/gpsd-3.21/.configured_68b329da9893e34099c7d8ad5cb9c940] Error 2
make[2]: Leaving directory '/tmp/openwrt/feeds/packages/utils/gpsd'
time: package/feeds/packages/gpsd/compile#17.07#2.29#18.78
    ERROR: package/feeds/packages/gpsd failed to build.
make[1]: *** [package/Makefile:116: package/feeds/packages/gpsd/compile] Error 1
make[1]: Leaving directory '/tmp/openwrt'
make: *** [/tmp/openwrt/include/toplevel.mk:230: package/gpsd/compile] Error 2
```

--- a/utils/gpsd/Makefile
+++ b/utils/gpsd/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=gpsd
 PKG_VERSION:=3.21
-PKG_RELEASE:=3
+PKG_RELEASE:=4
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=@SAVANNAH/$(PKG_NAME)
@@ -95,6 +95,7 @@ define Package/libgps/description
 endef
 
 SCONS_VARS += \
+	CFLAGS="$(TARGET_CFLAGS) $(EXTRA_CFLAGS) -L$(STAGING_DIR)/usr/lib" \
 	LINKFLAGS="$(TARGET_LDFLAGS)"
 
 SCONS_OPTIONS += \
@@ -116,7 +117,7 @@ SCONS_OPTIONS += \
 	implicit_link=no \
 	chrpath=no \
 	manbuild=no \
-	sysroot="$(STAGING_DIR)" \
+	sysroot="$(TOOLCHAIN_DIR)" \
 	target="$(TARGET_CROSS:-=)"
 
 define Build/InstallDev

